src/base/QXmppPushEnableIq.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2008-2020 The QXmpp developers
+ *
+ * Author:
+ *  Rober Märkisch
+ *  Linus Jahn
+ *  Jonah Brüchert
+ *
+ * Source:
+ *  https://github.com/qxmpp-project/qxmpp
+ *
+ * This file is a part of QXmpp library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include "QXmppPushEnableIq.h"
+
+#include "QXmppConstants_p.h"
+#include "QXmppDataForm.h"
+
+#include <QDebug>
+#include <QDomElement>
+
+class QXmppPushEnableIqPrivate : public QSharedData
+{
+public:
+    QString node;
+    QString jid;
+    QXmppPushEnableIq::Mode mode;
+    QXmppDataForm dataForm;
+};
+
+QXmppPushEnableIq::QXmppPushEnableIq()
+    : d(new QXmppPushEnableIqPrivate())
+{
+}
+
+QXmppPushEnableIq::~QXmppPushEnableIq() = default;
+
+QXmppPushEnableIq::QXmppPushEnableIq(const QXmppPushEnableIq &) = default;
+QXmppPushEnableIq &QXmppPushEnableIq::operator=(const QXmppPushEnableIq &) = default;
+
+///
+/// \brief returns the jid of the app server
+///
+QString QXmppPushEnableIq::jid() const
+{
+    return d->jid;
+}
+
+///
+/// \brief sets the jid of the app server
+///
+void QXmppPushEnableIq::setJid(const QString &jid)
+{
+    d->jid = jid;
+}
+
+///
+/// \brief returns the pubsub node on the app server used by the IQ
+///
+QString QXmppPushEnableIq::node() const
+{
+    return d->node;
+}
+
+///
+/// \brief set the pubsub note on the app server to be used by the IQ
+///
+void QXmppPushEnableIq::setNode(const QString &node)
+{
+    d->node = node;
+}
+
+///
+/// \brief returns the mode
+///
+QXmppPushEnableIq::Mode QXmppPushEnableIq::mode()
+{
+    return d->mode;
+}
+
+///
+/// \brief set whether the IQ should enable or disable push notifications
+///
+void QXmppPushEnableIq::setMode(QXmppPushEnableIq::Mode mode)
+{
+    d->mode = mode;
+}
+
+QXmppDataForm QXmppPushEnableIq::dataForm() const
+{
+    return d->dataForm;
+}
+
+void QXmppPushEnableIq::setDataForm(const QXmppDataForm &form)
+{
+    d->dataForm = form;
+}
+
+void QXmppPushEnableIq::parseElementFromChild(const QDomElement &element)
```

Comment:
Please add `/// \cond` here and `/// \endcond` after toXml() to ignore doxygen warnings

and please also in the header file